### DISCUSSION TITLE
Automated failover for replica clusters

### DISCUSSION THREAD

**Discussion by orient19** [cloudnative-pg/cloudnative-pg#1178] [General]
Hello!
Are you planning to do an Automated failover for [Replica clusters](https://cloudnative-pg.io/documentation/1.17/replica_cluster/)?
There are two kubernetes clusters (primary and reserve). I want to deploy a postgresql replica cluster on a standby kubernetes cluster. The documentation states: "Disabling replication is an irreversible operation: once replication is disabled and the designated primary is promoted to primary, the replica cluster and the source cluster will become two independent clusters definitively.". Is it possible to make the promotion of the designated primary to primary a reversible and automatic procedure?

**Comment by gbartolini** [comment]
I am preparing a blog article about this.

**Comment by orient19** [comment]
Hello! Thank you for your reply. I'll be waiting for your blog post.

**Comment by gbartolini** [comment]
@orient19 while trying to write the blog article, I noticed that what I had in mind does not currently work. I tried to properly shutdown the primary data centre, promote the other data centre and reconfigure the previous one to replicate from the new primary data centre. Unfortunately, it is not currently possible. This is primarily because until now we have not actually prioritized this use case.

At least, the operator should be able to resync a former primary to a designated primary. I will leave this ticket open for when we will start working on this.

**Comment by nmargaritis** [comment]
I am also interested in this use-case. Primarily because we are looking into cluster setup that crosses AZ/DC zones. So is failover of the operator between 2 independent k8s clusters something to be taken care of manually at the moment?

i reference this:
```
CloudNativePG does not perform any cross-cluster switchover or failover at the moment. Such operation must be performed manually or delegated to a multi-cluster/federated cluster aware authority. Each PostgreSQL cluster is independent from any other.
```

How would a usecase like the following be supported by cloudnative-gg when it comes to failovers at the moment.
- k8s cluster on AZ1 holding primary node of postgres operator
- k8s cluster on AZ2 holding standby node of postgres operator

Also is the following possible in the current setup?
- Could we have write delegation on standby side (via pgbouncer/pgpool II) so that both primary and standby can be active when it comes to reads and only have the active-standby setup on writes? Would the writer move if one cluster completely fails? (given that network access is configured)

**Comment by nmargaritis** [comment]
@gbartolini any updates on that?

**Comment by matthewhembree** [comment]
Reversibility would be good for orgs that test their disaster recovery processes.

**Comment by sxd** [comment]
This looks like a discussion and not an issue, turning into discussion
